### PR TITLE
User update, delete, unit tests

### PR DESCRIPTION
**Pull request by Jim3Things**
This change includes support for update & delete operations on users, restructures and extends the unit tests

### CODE REVIEW

File: internal/services/frontend/frontend.go [hunk 1]
```diff
@@ -161,6 +126,67 @@ func (he *HTTPError) Error() string {
     return he.Base.Error()
 }
 
+// +++ HTTPError specializations
+
+// ErrNoLoginActive indicates that the specified user is not logged into this session
+func NewErrNoLoginActive(name string) *HTTPError {
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/frontend.go:132]: The significance of the "New" prefix on this and the following errors?

File: internal/services/frontend/users.go [hunk 2]
```diff
@@ -319,34 +329,111 @@ func handlerUsersRead(w http.ResponseWriter, r *http.Request) {
...
+            return err
+        }
+
+        // Finally, get the current user definition and return it.  Again, protecting
+        // against the user getting deleted in another (racing) thread.
+        u, rev, err := dbUsers.Get(username)
+        if err != nil {
+            httpError(ctx, span, w, err)
+            return err
+        }
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/users.go:391]: If there is always (likely) to be a Get() after an update, perhaps the update should return the newly updated user record on success? That way we 
1. only need a single call,
2. get rid of a potential error case (update succeeds, get fails)

File: internal/services/frontend/frontend.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/frontend/frontend.go:132]: I was following a pattern that I'd seen - the NewXxx creates a new object for Xxx.  In this case, the ErrNoLoginActive is a specialization of HTTPError, which makes it a bit funky.

File: internal/services/frontend/users.go (hunk 2 above)
Comment (Jim3Things) [reply internal/services/frontend/users.go:391]: Yeah, I thought about that, and probably should go that way.

Comment (Jim3Things) [reply internal/services/frontend/users.go]: Fixed in this latest update.

File: internal/services/frontend/frontend.go (hunk 1 above)
Comment (MikeCJohnsonUk) [reply internal/services/frontend/frontend.go:132]: Hhmm, I find myself conflicted. I see the reasoning, but it still looks wrong somehow. I need to think on this to figoure out how I need to reconcile those two. But no reason to hold the checkin.

File: internal/services/frontend/users.go (hunk 2 above)
Comment (MikeCJohnsonUk) [reply internal/services/frontend/users.go]: Ok

**Pull request merged**